From e351a0f0b393be0cde8778af7c91f12bde624f92 Mon Sep 17 00:00:00 2001
From: Liang Chen <cl@rock-chips.com>
Date: Tue, 21 Nov 2017 17:16:50 +0800
Subject: [PATCH] ARM: dts: rockchip: do not access qos-ebc for rk3126b/c

rk3128 have ebc, but rk3126b/c have no ebc, so we need
manage compatibility for them.

Change-Id: I683f292b40603fe31a5eb4cee4e78a2d5d152626
Signed-off-by: Liang Chen <cl@rock-chips.com>
---
 arch/arm/boot/dts/rk3128.dtsi | 13 +++++++++++++
 arch/arm/boot/dts/rk312x.dtsi |  8 +-------
 2 files changed, 14 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/rk3128.dtsi b/arch/arm/boot/dts/rk3128.dtsi
index d567996f6ac3..977cf008d543 100644
--- a/arch/arm/boot/dts/rk3128.dtsi
+++ b/arch/arm/boot/dts/rk3128.dtsi
@@ -43,6 +43,11 @@
 / {
 	compatible = "rockchip,rk3128";
 
+	qos_ebc: qos@1012f080 {
+		compatible = "syscon";
+		reg = <0x1012f080 0x20>;
+	};
+
 	cru: clock-controller@20000000 {
 		compatible = "rockchip,rk3128-cru";
 		reg = <0x20000000 0x1000>;
@@ -52,6 +57,14 @@
 	};
 };
 
+&pd_vio {
+	pm_qos = <&qos_rga>,
+		 <&qos_ebc>,
+		 <&qos_iep>,
+		 <&qos_lcdc0>,
+		 <&qos_vip0>;
+};
+
 &vop_mmu {
 	rockchip,skip-mmu-read;
 };
diff --git a/arch/arm/boot/dts/rk312x.dtsi b/arch/arm/boot/dts/rk312x.dtsi
index e7ce8c6d8916..69735f0a161e 100644
--- a/arch/arm/boot/dts/rk312x.dtsi
+++ b/arch/arm/boot/dts/rk312x.dtsi
@@ -288,7 +288,7 @@
 			#size-cells = <0>;
 			status = "okay";
 
-			pd_vio@RK3128_PD_VIO {
+			pd_vio: pd_vio@RK3128_PD_VIO {
 				reg = <RK3128_PD_VIO>;
 				clocks = <&cru ACLK_RGA>,
 					 <&cru ACLK_LCDC0>,
@@ -309,7 +309,6 @@
 					 <&cru PCLK_MIPIPHY>,
 					 <&cru SCLK_VOP>;
 				pm_qos = <&qos_rga>,
-					 <&qos_ebc>,
 					 <&qos_iep>,
 					 <&qos_lcdc0>,
 					 <&qos_vip0>;
@@ -507,11 +506,6 @@
 		reg = <0x1012f000 0x20>;
 	};
 
-	qos_ebc: qos@1012f080 {
-		compatible = "syscon";
-		reg = <0x1012f080 0x20>;
-	};
-
 	qos_iep: qos@1012f100 {
 		compatible = "syscon";
 		reg = <0x1012f100 0x20>;
-- 
2.35.3

